zebra-consensus/src/checkpoint/tests.rs

```diff
@@ -280,8 +283,9 @@ async fn continuous_blockchain() -> Result<(), Report> {
             ready_verifier_service.call(block.clone()),
         );
 
-        /// SPANDOC: something here?
-        tokio::spawn(async move { verify_future.await });
```

Comment:
The way you have this currently setup it discards the `JoinHandle` returned by `tokio::spawn` which causes us to discard errors in the background and causing a possible race. I don't think this will cause the test to pass when it shouldn't, because all of the important logic for the verifier is done in `call`, so we shouldn't run into any race conditions, however we can't rely on this always being the case so we should ensure all of the futures we spawn are complete before exiting the test, this will also improve the diagnostics if we do encounter an error because we will log it with our own fancy error handling code instead of letting tokio log it in the background for us, which I'm not even sure it will always do.

Reply:
@yaahc thanks, great stuff. I think it was the missing part and will have this present for future tests.